src/directLine.ts

```diff
@@ -398,10 +405,33 @@ export class DirectLine implements IBotConnection {
         return once ? obs.take(1) : obs;
     }
 
+    setConnectionStatusFallback(
+        connectionStatusFrom: ConnectionStatus,
+        connectionStatusTo: ConnectionStatus,
+        maxAttempts = 5
+     ) {
+        let attempts = 0;
```

Comment:
If we initially set `attempts = 1`, then we can get rid of the `maxAttempts - 1` below.